presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcPageSink.java

```diff
@@ -159,6 +161,10 @@ else if (DATE.equals(type)) {
             long localMillis = getInstanceUTC().getZone().getMillisKeepLocal(DateTimeZone.getDefault(), utcMillis);
             statement.setDate(parameter, new Date(localMillis));
         }
+        else if (TIMESTAMP.equals(type)) {
+            long utcMillis = type.getLong(block, position);
+            statement.setTimestamp(parameter, new Timestamp(utcMillis));
```

Comment:
I would expect this to work only when
- `legacy_timestamp=true` and JVM zone is the same as user session zone
- `legacy_timestamp=false` and JVM zone is UTC

Can you add tests?